lib/ansible/plugins/connection/ssh.py

```diff
@@ -1122,7 +1122,9 @@ def reset(self):
             p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             stdout, stderr = p.communicate()
             status_code = p.wait()
-            if status_code != 0:
+            if status_code == 255 and ('No such file or directory' in stderr):
```

Comment:
Parsing the error message here is problematic (eg, think localized ssh). Maybe better to just repeat the check of `cp_path` if defined instead (ie, if we verify it's already in the right state because the file is missing, send the debug message and move on).

Reply:
Doh. I missed the check of cp_path above. I was using/looking at 2.6.5, and then hacked in this change in master without reading to see what else changed. The issue I was attempting to fix is already fixed in 2.7 with #43062